src/app/locationForm/number/LocationNumber.jsx

```diff
@@ -1,25 +1,113 @@
+import Button from '../../../components/button';
+import Input from '../../../components/input';
 
-const LocationNumberEdit = compose(
-  withProps({
-    headerText : 'What\'s this location\'s phone number?',
-    placeholderText : 'e.g. 646-909-4591',
-  })
-)(props => <FormEdit {...props} />)
+class LocationNumberEdit extends Component {
+  constructor(props) {
+    super(props);
+
+    const [areaCode, firstThree, lastFour] = props.value.number ? props.value.number.split('.') : ['','',''];
+    this.state = {
+      areaCode, 
+      firstThree, 
+      lastFour,
+      extension : props.value.extension || ''
+    };
+
+    this.onSubmit = this.onSubmit.bind(this);
+    Object.keys(this.state).forEach(k => {
+      this[`onChange_${k}`] = this.onChange.bind(this, k);
+    });
+  }
+
+  componentWillReceiveProps(props){
+    let o = {};
+    if(props.value.number !== this.props.value.number){
+      const [areaCode, firstThree, lastFour] = props.value.number ? props.value.number.split('.') : [];
+      o = {
+        areaCode, 
+        firstThree, 
+        lastFour
+      }
+    }
+
+    if(props.value.extension !== this.props.value.extension){
+      o.extension = props.value.extension;
+    }
+
+    if(Object.keys(o).length) this.setState(o);
+  }
+
+  onChange(key, event) {
+    this.setState({ [key]: event.target.value });
+  }
+
+  onSubmit() {
+    this.props.updateValue({
+      number : [this.state.areaCode, this.state.firstThree, this.state.lastFour].join('.'),
+      extension : this.state.extension
+    });
+    this.props.onSubmit();
+  }
+
+  render() {
+    return (
+      <div>
+        <Header>What's this location's phone number?</Header>
+          <div>
+            <Input
+              style={{width:'2em'}}
+              type="number"
+              value={this.state.areaCode}
+              onChange={this.onChange_areaCode}
+            />&nbsp;-&nbsp;
+            <Input
+              style={{width:'2em'}}
+              type="number"
+              value={this.state.firstThree}
+              onChange={this.onChange_firstThree}
+            />&nbsp;-&nbsp;
+            <Input
+              style={{width:'3em'}}
+              type="number"
+              value={this.state.lastFour}
+              onChange={this.onChange_lastFour}
+            />
+          </div>
+          <div>
+            ext.
+            <Input
+              type="number"
+              value={this.state.extension}
+              onChange={this.onChange_extension}
+            />
+          </div>
+        <Button onClick={this.onSubmit} primary className="mt-3">
+          OK
+        </Button>
+      </div>
+    );
+  }
+}
 
 const LocationNumberView = compose(
   withProps({
     topText : 'LOCATION PHONE NUMBER',
   })
-)(props => <FormView {...props} />)
+)(props => {
+  const value = props.value && props.value.number && `${props.value.number.replace(/\./g,'-')}${props.value.extension ? ` ext.${props.value.extension}` : ''}`;
```

Comment:
I assume this is just a temporary way of showing the number, and later we'll make it match the designs? (And also add support for the list of numbers shown in the [designs](https://www.figma.com/file/fdzJjvkSugVGnUa3zNdvMNsU/StreetTeam_source_of_truth?node-id=4%3A1290)?)

Reply:
At that point I suppose we'll also need the LocationNumberView to be a custom component.
That makes me think it probably makes sense to break this file down to split the container and presentational components, as even with just LocationNumberEdit inside it it's already doing quite a lot, and probably moreso once we have a proper LocationNumberView.

Reply:
Yes, exactly, this is a temporary view until we can make it better match the designs.